apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts

```diff
@@ -0,0 +1,231 @@
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+const addSharingSetPopover = (resLocals) => {
+  return html`
+    <form name="sharing-set-create" method="POST">
+      <input type="hidden" name="__action" value="unsafe_sharing_set_create">
+      <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}">
+
+      <div class="form-group mb-4">
+        <p class=form-text>
+          Enter the name of the sharing set you would like to create.
+        </p>
+      </div>
+      <div class=form-group>
+        <input class="form-control form-control-sm" type="text" name="sharing_set_name" required/>
+      <div>
+      <div class="text-right mt-4">
+        <button type="button" class="btn btn-secondary" onclick="$('#courseSharingSetAdd').popover('hide')">Cancel</button>
+        <button type="submit" class="btn btn-primary">Create Sharing Set</button>
+      </div>
+    </form>
+  `.toString();
+};
+
+const chooseSharingNamePopover = (resLocals) => {
+  return html`
+    <form name="choose-sharing-name" method="POST">
+      <input type="hidden" name="__action" value="unsafe_choose_sharing_name">
+      <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}">
+
+      <div class="form-group mb-4">
+        <p class=form-text>
+          Enter the sharing name you would like for your course.
+        </p>
+        <p><strong>Once you choose your course's sharing name, you will not be able to change it</strong>,
+          because doing so would break the assessments of other courses that have imported your questions.
+          It is recommended that you choose something short but descriptive. For example, if you're teaching
+          a calculus course at a university that goes by the abbreviation 'XYZ', then you could choose the sharing
+          name 'xyz-calculus'. Then other courses will import questions from your course with the syntax '@xyz-calculus/qid'.
+        </p>
+      </div>
+      <div class=form-group>
+        <input class="form-control form-control-sm" type="text" name="course_sharing_name" required/>
+      <div>
+      <div class="text-right mt-4">
+        <button type="button" class="btn btn-secondary" onclick="$('#chooseSharingName').popover('hide')">Cancel</button>
+        <button type="submit" class="btn btn-primary">Choose Sharing Name</button>
+      </div>
+    </form>
+  `.toString();
+};
+
+export const InstructorSharing = ({ sharing_name, sharing_token, sharing_sets, resLocals }) => {
```

Comment:
The properties here should be typed. See here for an example:

https://github.com/PrairieLearn/PrairieLearn/blob/b0c1ff1537778e441508980c7bf86644e9e0b76f/apps/prairielearn/src/ee/pages/institutionAdminGeneral/institutionAdminGeneral.html.ts#L13-L21

Reply:
Sure! When I first wrote this, it was JavaScript. So much has changed!

Reply:
done.